### PR TITLE
Media Picker: Fix the no results message position

### LINKED ISSUE #18840: Media library: 'No media matching your search' message is no visible
### Expected behavior
While searching for media in Media Library, `No media matching your search` message is displayed when no results are found.
Also, if the search is cancelled and the user tries to add a media file again, they will see all available media in the search screen.

### Actual behavior
When searching Media Library without results, `No media matching your search` message is not visible, but underneath the header.
Also, when the search is canceled by tapping the `Cancel` header button the `No media matching your search` message will be displayed if the user tries to add Media Library items again for any media block.

### Steps to reproduce the behavior
1. Create a new blog post
2. Add a Image block
3. Click `ADD IMAGE`
4. Click `WordPress Media Library`
5. Search for some string that will not return any results
6. ❌ `No media matching your search` message is not visible. _Expected: The message IS visible._
7. Click `Cancel` button in Navigation bar
8. Click `ADD IMAGE`
9. Click `WordPress Media Library`
10. ❌ `No media matching your search` message is visible. _Expected: Available images are displayed_

https://user-images.githubusercontent.com/42008628/172656786-618ba8ee-b2da-4bd2-bc8c-d6cc3c686f0d.MP4

### PR DESCRIPTION
**Pull request by diogo-lins**
Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/18840, https://github.com/wordpress-mobile/WordPress-iOS/issues/19514
WPMediaPicker: https://github.com/wordpress-mobile/MediaPicker-iOS/pull/399 

## Description
This PR is the union of the changes made on the opened PR to the `WPMediaPicker-iOS` pod and a minor fix on the `WordPress-iOS` project side.

### Issue on WPMediaPicker-iOS:
It was observed after the user started searching for some media and returned a no-result media screen, **_the message presented was not showing correctly when the keyboard popped up_**.

#### Reason:
When the keyboard pops up, the code changes the `collectionView` `contentInset` and, in sequence, changes the frame size and position of the empty view based on the new `contentInset` values.

bug_before_current_wpmediapicker

#### What changed on the WPMediaPicker pod?
Now we have a UIView container to receive the `emptyView` from the delegate. This container is attached in the `collectionView` using constraints to anchor. When the keyboard pops up, we only need to change the bottom constraint to update the container size, getting the expected result. Also, all views on the `WPMediaPickerViewController` are now attached to the view using the constraint auto layout.

fix_after_wpmediapicker_beta

#### What changes on the WP-iOS project?
Even using the new beta version of the Pod, in landscape mode, the message is showing behind the keyboard. This message belongs to `NoResultsViewController`, so a change was made to update the top constraint for landscape mode.

Simulator Screen Recording - iPhone 13 - 2022-11-03 at 22 47 39

## Testing instructions
1. Run the app
2. Switch to a site that has media uploaded
3. Open Blog Post and add an `IMAGE` block
4. Make sure that media are displayed correctly on the list
5. Opens the Keyboard
6. Search to get no results
7. Make sure that the message `No media matching your search` is displayed correctly in Portrait and Landscape modes, with and without the keyboard on the screen.

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Added test on `NoResultsViewControllerTests`

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by hassaanelgarem — COMMENTED** [review]
The issue is resolved when accessing the media picker from the editor, but it's still happening when accessing the media picker from My Site.

Steps to reproduce:
1. Open the app
2. Go to "My Site" -> "Menu"
3. Tap "Media"
4. Start typing in the search bar until no results label is displayed.
5. Rotate the device to landscape.
6. Notice that the label is cut off by the keyboard.

**Comment by diogo-lins** [comment]
> The issue is resolved when accessing the media picker from the editor, but it's still happening when accessing the media picker from My Site.
> 
> Steps to reproduce:
> 
> 1. Open the app
> 2. Go to "My Site" -> "Menu"
> 3. Tap "Media"
> 4. Start typing in the search bar until no results label is displayed.
> 5. Rotate the device to landscape.
> 6. Notice that the label is cut off by the keyboard.

Hey @hassaanelgarem 👋🏾, could you share a gif of the flow? I did a test here, and it seemed to work using the:
```
pod 'WPMediaPicker', :git => 'https://github.com/diogo-lins/MediaPicker-iOS.git', :branch => 'issue/18840-19514-emptyview-height-v2'
```
And the fix present on this PR.

See 👇🏾 

Simulator Screen Recording - iPhone 13 - 2022-11-07 at 10 51 12

**Comment by hassaanelgarem** [comment]
@diogo-lins Weird! That's not what I'm getting. I'm testing on an iPhone 14 Pro simulator.

Here you go:

No results issue - iPhone 14 Pro

**Comment by diogo-lins** [comment]
> @diogo-lins Weird! That's not what I'm getting. I'm testing on an iPhone 14 Pro simulator.
> 
> Here you go:
> 
> No results issue - iPhone 14 Pro

Yeah, you're right; I tested here too on iPhone 14 Pro simulator (my last gif was tested on the iPhone 13 simulator 😄), and we still have the bug; 
I'll change the approach, instead of manipulating the top constraint of the label, I'll apply constraints to centralize it, in this condition (`landscape + keyboard showing`). It will be centered for any device iPhone screen in this condition.

**Comment by diogo-lins** [comment]
Hey @hassaanelgarem 👋🏾 , the real reason for this still showing the bug on the `iPhone 14 and iOS 16`  is because of the following code:
```
override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?)
```
In the `NoResultsViewController` It's not being fired when the orientation changes, to call the `configureTitleViewConstraints()` in sequence.

I'm investigating why this happens only for the `Media` flow.

Curiously, it works for the older iOS versions, but in `iOS 16` it wasn't firing. It's weird 😅.

**Comment by diogo-lins** [comment]
Hey @hassaanelgarem 👋🏾 , I didn't find out why it is not calling the `traitCollectionDidChange` for `Media` flow 🤔 

I have a proposal that works for both scenarios, `Media` and `Editor`:
```
override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
    super.viewWillTransition(to: size, with: coordinator)
    setAccessoryViewsVisibility()
    DispatchQueue.main.async {
        self.configureTitleViewConstraints()
    }
}
 ```
Moving the block of code responsible for updating the UILabel constraints to the `viewWillTransition` function, since here we're also notified when the user changes the iPhone orientation. We also already have logic related to `portrait|landscape` to hide the image.

Let me know what you think about this before I push the commit. 🙇🏾

https://user-images.githubusercontent.com/20734091/200589253-d18a99a4-8dfc-4ff9-ad5e-416da644d980.mov

--------------------

(In the meanwhile, I'm still investigating why `traitCollectionDidChange` is not fired on `iOS 16` and in this specific flow when the orientation changes) 🔎

**Comment by hassaanelgarem** [comment]
@diogo-lins I think you can go ahead with your suggestion 👍

**Comment by diogo-lins** [comment]
> @diogo-lins I think you can go ahead with your suggestion 👍

Done! Already pushed here https://github.com/wordpress-mobile/WordPress-iOS/pull/19555/commits/bda2dc5641b9bffe2ce52a22adac7438fcfe15eb 😄

File: WordPress/WordPressTest/ViewRelated/Views/Controllers/NoResultsViewControllerTests.swift [hunk 1]
```diff
@@ -91,6 +92,20 @@ class NoResultsViewControllerTests: XCTestCase {
...
+        parentViewController.view.frame = CGRect(origin: .zero, size: parentViewSize)
+
+        // When
+        resultViewController.configureForNoSearchResults(title: Constants.noResultsText)
+        addResultViewControllerToParent()
+
+        // Then
+        XCTAssertEqual(resultViewController.titleLabel.text, Constants.noResultsText)
+        XCTAssertTrue(resultViewController.noResultsView.isHidden)
+        XCTAssertTrue(resultViewController.titleLabel.frame.width <= parentViewSize.width)
```
Comment (hassaanelgarem) [inline WordPress/WordPressTest/ViewRelated/Views/Controllers/NoResultsViewControllerTests.swift]:
```suggestion
        XCTAssertLessThanOrEqual(resultViewController.titleLabel.frame.width, parentViewSize.width)
```

**Review by hassaanelgarem — APPROVED** [review]
Works as described!

Only had one minor comment.

File: WordPress/WordPressTest/ViewRelated/Views/Controllers/NoResultsViewControllerTests.swift (hunk 1 above)
Comment (diogo-lins) [reply WordPress/WordPressTest/ViewRelated/Views/Controllers/NoResultsViewControllerTests.swift]: Done 😁 - https://github.com/wordpress-mobile/WordPress-iOS/pull/19555/commits/73e21748c19b0fafc31f127fa7f458e7bc980604

**Comment by mokagio** [comment]
Taking over merging this already approved PR to get ahead on the 21.2 code freeze scheduled for Monday.

**Pull request merged**